File: terminal_stronghold_calc.c
#include <stdio.h>
#include <math.h> // for the tan trig function and pi

int main()
{
    struct point
    {
        float x, z, f;
    };

    // declaring the points using structures to get values for the coordinates and angles
    struct point firstPoint, secondPoint;

    // declaring variables that will be the outputted result
    int x, z;

    // getting values for the coordinates and angles
    printf("Name your first x value: ");
    scanf("%f", &firstPoint.x);

    printf("Name your first z value: ");
    scanf("%f", &firstPoint.z);

    printf("Name your first f value: ");
    scanf("%f", &firstPoint.f);

    printf("Name your second x value: ");
    scanf("%f", &secondPoint.x);

    printf("Name your second z value: ");
    scanf("%f", &secondPoint.z);

    printf("Name your second f value: ");
    scanf("%f", &secondPoint.f);

    // solving for x and z coordinates of the stronghold
    // M_PI is the value of pi
    z = ((firstPoint.z * tan(-firstPoint.f * M_PI / 180)) - (secondPoint.z * tan(-secondPoint.f * M_PI / 180)) - firstPoint.x + secondPoint.x) / (tan(-firstPoint.f * M_PI / 180) - tan(-secondPoint.f * M_PI / 180));
    x = (z - firstPoint.z) * tan(-firstPoint.f * M_PI / 180) + firstPoint.x;

    // output results
    printf("\nThe stronghold is at (%i, %i).\n", x, z);

    return 0;
}

File: terminal_stronghold_calc.py
#!/usr/bin/env python3


def main():
    # importing the tangent trig function and pi value
    from math import tan, pi

    class points():
        x, z, f = 0, 0, 0

    firstPoint = points()
    secondPoint = points()

    try:
        # getting values for the coordinates and angles
        firstPoint.x = float(input("Name your first x value: "))
        firstPoint.z = float(input("Name your first z value: "))
        firstPoint.f = float(input("Name your first f value: "))
        secondPoint.x = float(input("Name your second x value: "))
        secondPoint.z = float(input("Name your second z value: "))
        secondPoint.f = float(input("Name your second f value: "))
    except ValueError:
        # runs if the try block returns an error and ends the program
        # the only error that can occur in the try block:
        #   if the input cannot be converted into a float (ex: if the input is not a number)
        print("That is not a number.")
    else:
        # runs if the try block runs successfully
        # solving for x and z coordinates of the stronghold
        z = ((firstPoint.z * tan(-firstPoint.f * pi / 180)) -
             (secondPoint.z * tan(-secondPoint.f * pi / 180)) - firstPoint.x +
             secondPoint.x) / (tan(-firstPoint.f * pi / 180) -
                               tan(-secondPoint.f * pi / 180))

        x = (z - firstPoint.z) * tan(-firstPoint.f * pi / 180) + firstPoint.x

        # output results
        print(f"\nThe stronghold is at {round(x), round(z)}.")


if __name__ == "__main__":
    main()
